cli: rpc time parsing

gloriousCode opened the pull request:
# PR Description
I noticed this when parsing time for #1091 that the start end times looked more funky than I'd like them to be. Turns out the implementation of `negateLocalOffset` was off. 

When running `gethistoriccandles` it would be returning candle data hours outside of my expected timeframe. Its harder to measure since `gethistoriccandles` truncates the time, but it was well over 11 hours + truncated time. I was expecting candle data up to late 22nd Dec/early 23rd Dec. Instead I was getting 1pm on the 22nd. See below:
```
 "start": "2022-12-20 21:00:00",
 "end": "2022-12-22 13:00:00",
```

where now it is in line with my expectations where it is in line with the truncated time
```
 "start": "2022-12-21 08:00:00",
 "end": "2022-12-23 00:00:00",
```

This also matches `negateLocalOffsetTS`'s implementation

## Type of change
- [x] Bug fix (non-breaking change which fixes an issue)


## How has this been tested
Printing times in `commands.go` and matching them up in `rpcserver.go`
I was doing [these changes via gist](https://gist.github.com/gloriousCode/7261f4e5311d95799aa70a30ac256f7a)

thrasher- commented on cmd/gctcli/helpers.go:
```diff
@@ -45,9 +45,7 @@ func closeConn(conn *grpc.ClientConn, cancel context.CancelFunc) {
-	loc := time.FixedZone("", -offset)
-
-	return t.In(loc).Format(common.SimpleTimeFormat)
+	return t.Add(-time.Duration(offset) * time.Second).Format(common.SimpleTimeFormat)
```
How about this?

```suggestion
	return t.In(time.UTC).Format(common.SimpleTimeFormat)
}

```

gloriousCode replied:
I could have sworn that there was a reason why we didn't do this........ But its matching
cmd:
```
2022-12-23 15:59:00 +1100 AEDT
```
rpcserver
```
 2022-12-23 15:59:00 +1100 AEDT
```

# 🌞

shazbert reviewed (approved):
tACK but a suggestion below: 
```diff
diff --git a/cmd/gctcli/helpers.go b/cmd/gctcli/helpers.go
index b23eb8794..e8412ebeb 100644
--- a/cmd/gctcli/helpers.go
+++ b/cmd/gctcli/helpers.go
@@ -35,18 +35,14 @@ func closeConn(conn *grpc.ClientConn, cancel context.CancelFunc) {
        }
 }

-// negateLocalOffset helps negate the offset of time generation
-// when the unix time gets to rpcserver, it no longer is the same time
-// that was sent as it handles it as a UTC value, even though when
-// using starttime it is generated as your local time
-// eg 2020-01-01 12:00:00 +10 will convert into
-// 2020-01-01 12:00:00 +00 when at RPCServer
-// so this function will minus the offset from the local sent time
-// to allow for proper use at RPCServer
+// negateLocalOffset converts a time value to a string in UTC format, correcting
+// for any local time offset.
 func negateLocalOffset(t time.Time) string {
-       return t.In(time.UTC).Format(common.SimpleTimeFormat)
+       return t.UTC().Format(common.SimpleTimeFormat)
 }

+// negateLocalOffsetTS converts a time value to a *timestamppb.Timestamp in UTC
+// format, correcting for any local time offset.
 func negateLocalOffsetTS(t time.Time) *timestamppb.Timestamp {
-       return timestamppb.New(t.In(time.UTC))
+       return timestamppb.New(t.UTC())
 }
```

thrasher- commented:
One other thing I noticed is that:

1) Our presentation of time variables uses time.Now() which will default to the local time (this is perfect)
2) However, since we use time.Parse with no timezone, this translates a supplied local time into a UTC time which is incorrect. (Example here: https://github.com/gloriousCode/gocryptotrader/blob/great-mate-negate/cmd/gctcli/commands.go#L4136). As a result, an improvement would be to switch all the time.Parses to `time.ParseInLocation` with `time.Local` as the last param to be consistent with our displayed user local times. Then negateOffset will ensure correct conversion to UTC time before usage with the RPC server

gloriousCode commented:
I've used `ParseInLocation` to parse the `time.Now()`'s in the CLI properly. The implementation of `timestamppb.New` uses Unix() so there was no need for `negateLocalOffsetPb(...)` and so I've removed it. I've also changed the timestamp parsing in the CLI and RPCserver side to include the timezone so that it can be parsed accurately and so I've removed `negateLocalOffset(...)`

thrasher- reviewed (approved):
Changes ACK, nice stuff!

shazbert reviewed (approved):
parsing works, looking good!
